### PR TITLE
Deserialize data

### PR DESCRIPTION
**Pull request by mpapierski**
This is required to implement althea-mesh/clarity#14 where test fixtures contains raw "RLP" data and it should be compared both ways i.e. compare deserialized RLP with the "filler" values (svr, data, value, gas, etc.), and serialize RLP based on the filler values (this is done as serialization works well).

There are many strange corner cases, and rust's checked arithmetic made complicated to handle few cases, but I believe it would work for our purpose. 

There are quite a few test cases for decoder including wrong & valid serialized Eth transactions.

### CODE REVIEW

File: src/de.rs [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+            Ok(s)
+        } else {
+            Err(Error::ExpectedString)
+        }
+    }
+
+    fn parse_bytes(&mut self) -> Result<&'de [u8]> {
+        let res = rlp::decode_length(&self.input)?;
+        if res.expected_type == ExpectedType::StringType {
+            // let s = str::from_utf8(&self.input[res.offset..res.offset + res.length])
```
Comment (jkilpatr) [inline src/de.rs]: why is this left commented out here?

**Review by jkilpatr — APPROVED** [review]
looks good, only one question

**Review by drozdziak1 — APPROVED** [review]

File: src/de.rs (hunk 1 above)
Comment (mpapierski) [reply src/de.rs]: Going to remove this one. A lot of trial and error I guess.
